Source/Scene/Model.js

```diff
@@ -1454,6 +1474,371 @@ define([
+
+    function stringifyArray(array) {
+        return array.map(function(val) {
+            return val.toFixed(20);
+        });
+    }
+
+    // Makes a copy of the given material, its technique, and its program,
+    // and adds them to the gltf hierarchy with new ids
+    // returns the id of the new material
+    function splitMaterial(model, materialId) {
+        var gltf = model.gltf;
+        var num = 0;
+
+        var material = gltf.materials[materialId];
+        var newMaterialId;
+
+        // Find an available name for the new material
+        while (!defined(newMaterialId) ||
+               defined(gltf.materials[newMaterialId])) {
+            newMaterialId = materialId + '-' + num;
+            num++;
+        }
+
+        var newMaterial = clone(material);
+        newMaterial.technique = splitTechnique(model, material.technique);
+        gltf.materials[newMaterialId] = newMaterial;
+        return newMaterialId;
+    }
+
+    // Makes a copy of the given technique and its program,
+    // and adds them to the gltf hierarchy with new ids
+    // returns the id of the new technique
+    function splitTechnique(model, techniqueId) {
+        var gltf = model.gltf;
+        var num = 0;
+
+        var technique = gltf.techniques[techniqueId];
+        var newTechniqueId;
+
+        // Find an available name for the new technique
+        while (!defined(newTechniqueId) ||
+               defined(gltf.techniques[newTechniqueId])) {
+            newTechniqueId = techniqueId + '-' + num;
+            num++;
+        }
+
+        var newTechnique = clone(technique);
+        newTechnique.program = splitProgram(model, technique.program);
+        gltf.techniques[newTechniqueId] = newTechnique;
+        return newTechniqueId;
+    }
+
+    // Makes a copy of the given program
+    // and adds it to the gltf hierarchy with a new id
+    // returns the id of the new program
+    function splitProgram(model, programId) {
+        var gltf = model.gltf;
+        var num = 0;
+
+        var program = gltf.programs[programId];
+        var newProgramId;
+
+        // Find an available name for the new program
+        while (!defined(newProgramId) ||
+               defined(gltf.programs[newProgramId])) {
+            newProgramId = programId + '-' + num;
+            num++;
+        }
+
+        var newProgram = clone(program);
+        gltf.programs[newProgramId] = newProgram;
+        return newProgramId;
+    }
+
+    function getAllPrimitives(model) {
+        var gltf = model.gltf;
+        var allPrimitives = [];
+
+        for (var meshId in gltf.meshes) {
+            if (gltf.meshes.hasOwnProperty(meshId)) {
+                var mesh = gltf.meshes[meshId];
+                var primitives = mesh.primitives;
+                if (defined(primitives)) {
+                    for (var i = 0; i < primitives.length; i++) {
+                        var primitive = primitives[i];
+                        allPrimitives.push(primitive);
```

Comment:
Probably write this as:
```javascript
allPrimitives.push(primitives[i]);
```
See "Do not create a local variable that is used only once unless it significantly improves readability" in https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/CodingGuide#basic-code-construction